In [1]:
import pandas as pd
from shortcuts import data_loader
import plotly.express as px
import plotly.graph_objects as go

In [2]:
filename = "1. AssignmentBusinessQuant.csv"
folders = ("Results",)
data = pd.read_csv(data_loader.data_path(filename, *folders))
data.head()

Item  Year  Sales  Sales Last Year  Sales Growth %
0  Aloe Vera  2010     27              NaN             NaN
1  Aloe Vera  2011     31             27.0            14.8
2  Aloe Vera  2012     99             31.0           219.4
3  Aloe Vera  2013     10             99.0           -89.9
4  Aloe Vera  2014     60             10.0           500.0

In [3]:
px.bar(data_frame=data[data.Year==2019], x="Item", y="Sales", title="Which item got the most sales in the year 2019?")

It is very clear that in the `year 2019` the `most sales` is `from Carrot`.

In [4]:
fig = px.line(data_frame=data, x="Year", y="Sales", color="Item", title="Sales Per Year of Each Item")
traces = fig.select_traces()
settings = next(traces)
settings['visible'] = True
for settings in traces:
    settings['visible'] = 'legendonly'
fig.show()

In [5]:
yearly_data = data.groupby(["Year"]).agg({
    "Sales": "sum",
})
yearly_data["Sales Last Year"] = yearly_data.Sales.shift()
yearly_data["Sales Growth %"] = (yearly_data["Sales"] / yearly_data["Sales Last Year"] - 1) * 100
yearly_data

Sales  Sales Last Year  Sales Growth %
Year                                        
2010   1061              NaN             NaN
2011   1148           1061.0        8.199811
2012   1203           1148.0        4.790941
2013   1034           1203.0      -14.048213
2014    962           1034.0       -6.963250
2015   1198            962.0       24.532225
2016   1151           1198.0       -3.923205
2017   1145           1151.0       -0.521286
2018    974           1145.0      -14.934498
2019   1103            974.0       13.244353

In [6]:
px.bar(data_frame=yearly_data, x=yearly_data.index, y="Sales Growth %", color="Sales", title="Sales Growth % Per Year")

We can conclude that in the `Year 2018` the company was in the `height loss` with `sales growth of -14.9%` and `total sales of 974`. And in the `year 2015` we had `sales growth of 24.5%` with `sales of 1198` which is the `height growth` of the company.

In [7]:
px.line(data_frame=yearly_data, x=yearly_data.index, y='Sales', title="Sales Per Year")

Sales were at the `peek in the Year 2012` with `sales of 1203` and the sales were at the `bottom in the year 2014` with `sales of 962`.

In [8]:
px.bar(data_frame=data, x="Item", y="Sales", color="Year", title="Each Items Sales")

Item having `most sales is Cauliflower` and `lowest sales is Brocolli`